pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -0,0 +1,137 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+// /subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/deployments/{deploymentName}/preflight?api-version={api-version}
+func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	var header http.Header
+	var b []byte
+
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+
+	resources, err := unmarshalRequest(body)
+	if err == nil {
```

Comment:
what happens when err is not nil?

Reply:
This is just personal preference, but it might be nicer to go
```
if err != nil {
		b = marshalValidationResult(api.ValidationResult{
			Status: api.ValidationStatusFailed,
			Error: &api.ManagementErrorWithDetails{
				Message: to.StringPtr(err.Error()),
			},
		})
		reply(log, w, header, b, statusCodeError(http.StatusOK))
}
```
And then do your `for` loop after, separately. It would help flatten the code out a little